source/is-literal.d.ts

```diff
@@ -0,0 +1,38 @@
+import type {Primitive} from './primitive';
+import type {Numeric} from './numeric';
+import type {IsNever, IsNotFalse} from './internal';
+
+/** @link https://stackoverflow.com/a/52806744/10292952 */
+type LiteralCheck<T, LiteralType extends Primitive> = (
+	IsNever<T> extends false // Must be wider than `never`
+		? [T] extends [LiteralType] // Must be narrower than `LiteralType`
+			? [LiteralType] extends [T] // Cannot be wider than `LiteralType`
+				? false
+				: true
+			: false
+		: false
+);
```

Comment:
Once #564 lands, this will need to be updated to `import type {IsNever} from './is-never';`